### PR TITLE
GAPI (fluid): optimization of Separable filter

### PR DESCRIPTION
**Pull request by elatkin**

### CODE REVIEW

File: modules/gapi/src/backends/fluid/gfluidimgproc_func.simd.hpp [hunk 1]
```diff
@@ -309,207 +345,381 @@ void run_yuv2rgb_impl(uchar out[], const uchar in[], int width, const float coef
...
+// this variant not using buf[] appears 15% faster than reference any-2-float code below
+template<bool noscale, typename SRC>
+static void run_sepfilter3x3_any2float(float out[], const SRC *in[], int width, int chan,
+                                       const float kx[], const float ky[], int border,
+                                       float scale, float delta)
 {
-    float ky0 = ky[0],
-          ky1 = ky[1],
-          ky2 = ky[2];
+    const int length = width * chan;
```
Comment (alalek) [inline modules/gapi/src/backends/fluid/gfluidimgproc_func.simd.hpp]: What is about cases `length` < `nlanes` ?

Comment (elatkin) [reply modules/gapi/src/backends/fluid/gfluidimgproc_func.simd.hpp]: The *reference* code would process such small cases.

**Comment by elatkin** [comment]
Performance on i5-6600 as ratio vs original OpenCV kernel
(ratio above 1 means GAPI outperforms original OCV kernel)

| Filter           | Image |  In  | Out   | Ratio   | Speed-up |
|----------------|----------|-----|--------|---------|---------------|
| Blur             | 1080p |       | 8U    | 1.02    | 17  times  |
|                    |            |       | 16S  | 1.07    | 40.7          |
|                    |            |       | 32F  | 2.15    | 25.2          |
| Box             | 1080p |       | 8U    | 1.04    | 17.3          |
|                    |            |       | 16S  | 0.99    | 42             |
|                    |            |       | 32F  | 1.99    | 22.8          |
| Gauss Blur | 1080p  |      | 8U    | 0.96    | 24.2          |
|                    |            |       | 16S  | 1.63    | 22.8          |
|                    |            |       | 32F  | 0.99    | 12.3          |
| Sep. Filter   | 1080p | 8U | 8U    | 13.6    | 37             |
|                    |            |       | 16S  | 5.55    | 22.4          |
|                    |            |       | 32F  | 4.07    | 14.4          |
|                    |            | 16S | 16S | 3.46    | 22            |
|                    |            |        | 32F | 2.43    | 14.6         |
|                    |            | 32F | 32F | 2.03    | 11.9         |

Sobel filter was optimized with my previous PR #13070 and #13133

|  Filter          | Image |  In  |  Out  | Ratio   | Speed-up  |
|----------------|---------|------|--------|----------|---------------|
| Sobel 3x3   | 1080p | 8U  | 8U    | 14.05  | 1.02           |
|                    |            |       | 16S  | 0.8      | 1.15           |
|                    |            |       | 32F  | 2.99    | 0.99           |
|                    |            | 16S | 16S | 1.8      | 1.02           |
|                    |            |        | 32F | 1.34    | 0.97           |
|                    |            | 32F | 32F | 0.88    | 1.07           |

**Comment by elatkin** [comment]
@alalek  
Could you please merge it?

**Pull request merged**